src/keyring/KeyringController.test.ts

```diff
@@ -119,7 +123,29 @@ describe('KeyringController', () => {
     expect(initialState).not.toBe(currentState);
   });
 
+  it('should restore same vault if old seedWord is used', async () => {
+    const currentSeedWord = await keyringController.exportSeedPhrase(password);
+    const currentState = await keyringController.createNewVaultAndRestore(
+      password,
+      currentSeedWord,
+    );
+    expect(initialState).toStrictEqual(currentState);
+  });
+
+  it('should throw error if creating new vault and restore without password', async () => {
+    await expect(async () => {
+      await keyringController.createNewVaultAndRestore('', seedWords);
+    }).rejects.toThrow('Invalid password');
```

Comment:
[nit] `expect` also takes a promise directly, so if you like, you don't need the function here, you can shorten this to:

``` suggestion
    await expect(
      keyringController.createNewVaultAndRestore('', seedWords)
    ).rejects.toThrow('Invalid password');
```

That said, I see this pattern being used frequently in this file, so not a big deal — just a shortcut going forward.